operators/pm_from_dataset.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2022 Juan Pablo Tosso and the OWASP Coraza contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package operators
+
+import (
+	"fmt"
+
+	"github.com/corazawaf/coraza/v3"
+	ahocorasick "github.com/petar-dambovaliev/aho-corasick"
+)
+
+// TODO according to coraza researchs, re2 matching is faster than ahocorasick
+// maybe we should switch in the future
+// pmFromDataset is always lowercase
+type pmFromDataset struct {
+	matcher ahocorasick.AhoCorasick
+}
+
+func (o *pmFromDataset) Init(options coraza.RuleOperatorOptions) error {
+	data := options.Arguments
+	dataset, ok := options.Datasets[data]
+	if !ok {
+		return fmt.Errorf("Dataset %q not found", data)
+	}
+	builder := ahocorasick.NewAhoCorasickBuilder(ahocorasick.Opts{
+		AsciiCaseInsensitive: true,
+		MatchOnlyWholeWords:  false,
+		MatchKind:            ahocorasick.LeftMostLongestMatch,
+		DFA:                  true,
+	})
+
+	// TODO this operator is supposed to support snort data syntax: "@pmFromDataset A|42|C|44|F"
```

Comment:
nit: I think this TODO is misleading, snort syntax would end up being inside the Dataset, wouldn't it?

Reply:
Yep, we should keep it only for PM.

Regarding duplications, it depends on the operator, the dictionary will store everything case-sensitive. The aho-corasick implementation will read it case-insensitive and might probably remove duplicates. 

Still, a duplicate shouldn't affect results, just performance.